### DISCUSSION TITLE
Microservices Communication

### DISCUSSION THREAD

**Discussion by svipandi** [ChilliCream/hotchocolate#2838] [Q&A]
We have developed multiple microservices using .net core. All microservice have direct communication. We want to disconnect the relationship. I want to implement like apollo federation. Services should be loosely coupled. It needs to communicate only with the gateway. How we can achieve the same thing using hot chocolate? is there any example? so that I can move all my services under a single umbrella.

@michaelstaib  Please provide your input.

**Comment by PascalSenn** [comment]
Hi @svipandi  
Checkout this example: 
https://github.com/ChilliCream/hotchocolate-examples/tree/master/misc/Stitching/federated-with-hot-reload 

Docs are coming in this PR:
https://github.com/ChilliCream/hotchocolate/pull/2780

**Reply by shenbagapandiyan** [reply]
Hi @PascalSenn 
Thank you for your response.
I am trying to run redis-stitching in my local by the provided image. It's not running my local. Could you help with this?.
![image](https://user-images.githubusercontent.com/56395553/104152336-425ea280-5405-11eb-9733-2f688c2873c2.png)

**Reply by PascalSenn** [reply]
Never seen this warning

**Comment by shenbagapandiyan** [comment]
Hi @PascalSenn  @michaelstaib 
I have configured all the services in my local. All individual services are working fine. Gateway alone couldn't display the schema. Please help on this.

**Reply by PascalSenn** [reply]
What is not working?

**Reply by shenbagapandiyan** [reply]
I have tried two examples
1) Federated with pull -  it's working only with products and reviews. I can able to see the merged schema from the gateway. When I try to add accounts or inventory services its starts closing immediately once it's started.

2) Federated with hot reload - schema has been created in Redis.  But application starts closing immediately once it's started.

**Reply by michaelstaib** [reply]
BTW ... I have an updated example here:
https://github.com/ChilliCream/hotchocolate-examples/tree/demo/misc/Stitching/federated-with-hot-reload

**Reply by michaelstaib** [reply]
Its in the demo branch and you can do ....

```bash
dotnet tool restore
dotnet the run
```
